### PR TITLE
HPCC-18093 WsESDLConfig getDef should accept name without version

### PR DESCRIPTION
**Pull request by rpastrana**
- Adds name and version fields for lookup by name only or name.ver
- ID can contain name only

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@afishbeck please take a look

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 1]
```diff
@@ -157,10 +157,48 @@ bool isESDLDefinitionBound(const char * esdldefname, int version)
     return isESDLDefinitionBound(id);
 }
 
+
+void fetchESDLDefinitionFromDaliByNameOnly(const char * name, StringBuffer & def)
+{
+    if (!name || !*name)
+        throw MakeStringException(-1, "Unable to fetch ESDL Service definition information, definition name is not available");
```
Comment (afishbeck) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: Should be assigning error codes for these exceptions.  Can add them to the same header EclWatch uses.

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 2]
```diff
@@ -157,10 +157,48 @@ bool isESDLDefinitionBound(const char * esdldefname, int version)
...
+
+    DBGLOG("ESDL Binding: Fetching ESDL Definition from Dali based on name: %s ", name);
+
+    Owned<IRemoteConnection> conn = querySDS().connect(ESDL_DEFS_ROOT_PATH, myProcessSession(), RTM_LOCK_READ, SDS_LOCK_TIMEOUT_DESDL);
+    if (!conn)
+       throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", ESDL_DEFS_ROOT_PATH);
+
+    conn->close(false);//release lock right away
```
Comment (afishbeck) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: This seems a little strange.  It looks like EsdlConfig is the only service that does it quite this way.. is it kosher?  @jakesmith can you comment?

**Review by afishbeck — COMMENTED** [review]
@rpastrana a couple comments.

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp (hunk 2 above)
Comment (rpastrana) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: this was a cary-over, code was changed in Master branch. will fix here as well

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp (hunk 1 above)
Comment (rpastrana) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: Good point, as of now this is pervasive in this class and others, a dedicated task should be opened to tackle this issue.

**Comment by rpastrana** [comment]
@afishbeck one more review

**Review by afishbeck — APPROVED** [review]
@rpastrana looks good

**Comment by rpastrana** [comment]
@richardkchapman please merge

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp (hunk 1 above)
Comment (rpastrana) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: https://track.hpccsystems.com/browse/HPCC-18199

**Pull request merged**